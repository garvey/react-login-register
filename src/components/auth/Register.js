import React from 'react';
import { Link } from 'react-router-dom';
import { FiAlertCircle } from 'react-icons/fi';
import useForm from 'react-hook-form';
import Logo from '../../images/logo.png';

import {
  Col,
  FormGroup,
  Label,
  Button,
  Row,
  Container,
  Form,
  Card,
  CardBody,
  Alert
} from 'reactstrap';

function Register(props) {
  const { register, handleSubmit, errors } = useForm();
  const onSubmit = res => {
    console.log('Register Data:', res);
  };

  return (
    <Container>
      <Row>
        <Col
          xs={12}
          md={{ size: 10, offset: 1 }}
          lg={{ size: 8, offset: 2 }}
          className="my-auto mx-auto"
        >
          <Form onSubmit={handleSubmit(onSubmit)}>
            <div className="text-center pt-4 pb-4">
              <Link to="/">
                <img src={Logo} alt="Company Logo" className="img-fluid logo" />
              </Link>
            </div>
            <Card className="register">
              <CardBody>
                <Row>
                  <Col xs={12} sm={6}>
                    <h2>Register</h2>
                  </Col>
                  <Col xs={12} sm={6} className="align-end">
                    <small className="float-sm-right">
                      Have an account? <Link to="/login">Sign in here</Link>
                    </small>
                  </Col>
                </Row>
                <Row className="divider">
                  <Col xs={12} sm={4}>
                    <h5>Personal Details</h5>
                  </Col>

                  <Col xs={12} sm={6}>
                    <FormGroup>
                      <Label>Full Name</Label>
                      <input
                        name="name"
                        ref={register({
                          required: 'Please Enter your name'
                        })}
                      />
                    </FormGroup>
                    {errors.name && (
                      <Alert color="info">
                        <FiAlertCircle /> Please enter your name
                      </Alert>
                    )}
                  </Col>
                </Row>

                <Row className="divider">
                  <Col xs={12} sm={4}>
                    <h5>Login Details</h5>
                  </Col>

                  <Col xs={12} sm={6}>
                    <FormGroup>
                      <Label>Email</Label>
                      <input
                        name="email"
                        ref={register({
                          required: 'Required email address',
                          pattern: {
                            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
                            message: 'invalid email address'
                          }
                        })}
                      />
                    </FormGroup>

                    {errors.email && errors.email.message && (
                      <Alert color="info">
                        <FiAlertCircle /> Please enter a valid email address
                      </Alert>
                    )}

                    <FormGroup>
                      <Label>Password</Label>
                      <input
                        name="password"
                        type="password"
                        ref={register({ required: true, maxLength: 10 })}
                      />
                    </FormGroup>
                    {errors.password && (
                      <Alert color="info">
                        <FiAlertCircle /> Please enter your password
                      </Alert>
                    )}
                  </Col>
                </Row>

                <Row className="divider footer">
                  <Col>
                    <div className="btn-group float-right">
                      <Button color="success" outline block type="submit">
                        Register
                      </Button>
                    </div>
                  </Col>
                </Row>
              </CardBody>
            </Card>
          </Form>
        </Col>
      </Row>
    </Container>
  );
}

export default Register;

// function Register(props) {
//   return (
// class Register extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       name: '',
//       email: '',
//       password: '',
//       error: false,
//       captcha: ''
//     };
//   }

//   handleSubmit = event => {

//   };

//   render() {
//     return (
//       <Container>
// <Col
//   xs={12}
//   md={{ size: 10, offset: 1 }}
//   lg={{ size: 8, offset: 2 }}
//   className="my-auto mx-auto"
// >
//           <div className="text-center pt-4 pb-4">
//             <Link to="/">
//               <img
//                 src={DenningsLogo}
//                 alt="Dennings Logo"
//                 className="responsive-image"
//               />
//             </Link>
//           </div>

//           <Card className="register">
//             <CardBody>
//               <Form onSubmit={this.handleSubmit}>
// <Row>
//   <Col xs={12} sm={6}>
//     <h2>Register</h2>
//   </Col>
//   <Col xs={12} sm={6} className="align-end">
//     <small className="float-sm-right">
//       Have an account? <Link to="/login">Sign in here</Link>
//     </small>
//   </Col>
// </Row>

// <Row className="divider">
//   <Col xs={12} sm={4}>
//     <h5>Personal Details</h5>
//   </Col>

//   <Col xs={12} sm={6}>
//     <FormGroup>
//       <Label for="name">Full Name</Label>
//       <Input
//         type="text"
//         name="name"
//         id="name"
//         placeholder="Enter your name..."
//         onChange={this.handleChange}
//         value={this.state.name}
//         required
//       />
//     </FormGroup>
//   </Col>
// </Row>

//                 <Row className="divider">
// <Col xs={12} sm={4}>
//   <h5>Login Details</h5>
// </Col>

//                   <Col xs={12} sm={6}>
//                     <FormGroup>
//                       <Label for="email">Email</Label>
//                       <Input
//                         type="email"
//                         name="email"
//                         id="email"
//                         placeholder="Enter your email..."
//                         onChange={this.handleChange}
//                         value={this.state.email}
//                       />
//                     </FormGroup>
//                     {this.state.error === true && (
//                       <p className="text-info">
//                         <FiAlertCircle /> This email address is already in use.
//                       </p>
//                     )}

//                     <FormGroup>
//                       <Label for="password">Password</Label>
//                       <Input
//                         type="password"
//                         name="password"
//                         id="password"
//                         onChange={this.handleChange}
//                         value={this.state.password}
//                         required
//                       />
//                     </FormGroup>

//                   </Col>
//                 </Row>
//                 <Row className="divider footer">
//                   <Col>
//                     <div className="btn-group float-right">
//                       <Button
//                         color="success"
//                         block
//                         type="submit"

//                       >
//                         Register
//                       </Button>
//                     </div>
//                   </Col>
//                 </Row>
//               </Form>
//             </CardBody>
//           </Card>
//         </Col>
//       </Container>
//     );
//   }
// }

// export default Register;
