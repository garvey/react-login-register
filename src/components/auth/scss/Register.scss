$form-footer-color: #ecf1f7;

.logo {
  max-width: 200px;
}

.register {
  border-top: 5px solid $blue;

  input {
    display: block;
    width: 100%;
    border: 1px solid #ced4da;
    border-radius: 0.25em;
    height: calc(1.5em + 0.75rem + 2px);
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
  }

  .form-check {
    padding-left: 0;
  }

  .footer {
    background: $form-footer-color;
    margin-left: -20px;
    margin-right: -20px;
  }

  .card-body {
    padding-bottom: 0;
  }

  .warning-container {
    margin: 0 auto;
  }
}
