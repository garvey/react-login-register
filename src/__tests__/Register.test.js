import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Register from '../components/auth/Register';

it('Should render Component', () => {
  const wrapper = mount(
    <MemoryRouter>
      <Register />
    </MemoryRouter>
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
