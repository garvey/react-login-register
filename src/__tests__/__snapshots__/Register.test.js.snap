// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Component 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Register>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col xs={12} md={{...}} lg={{...}} className=\\"my-auto mx-auto\\" tag=\\"div\\" widths={{...}}>
                <div className=\\"my-auto mx-auto col-12 col-md-10 offset-md-1 col-lg-8 offset-lg-2\\">
                  <Form onSubmit={[Function]} tag=\\"form\\">
                    <form onSubmit={[Function]} className=\\"\\">
                      <div className=\\"text-center pt-4 pb-4\\">
                        <Link to=\\"/\\">
                          <LinkAnchor href=\\"/\\" navigate={[Function: navigate]}>
                            <a href=\\"/\\" onClick={[Function: onClick]}>
                              <img src=\\"logo.png\\" alt=\\"Company Logo\\" className=\\"img-fluid logo\\" />
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                      <Card className=\\"register\\" tag=\\"div\\">
                        <div className=\\"register card\\">
                          <CardBody tag=\\"div\\">
                            <div className=\\"card-body\\">
                              <Row tag=\\"div\\">
                                <div className=\\"row\\">
                                  <Col xs={12} sm={6} tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col-12 col-sm-6\\">
                                      <h2>
                                        Register
                                      </h2>
                                    </div>
                                  </Col>
                                  <Col xs={12} sm={6} className=\\"align-end\\" tag=\\"div\\" widths={{...}}>
                                    <div className=\\"align-end col-12 col-sm-6\\">
                                      <small className=\\"float-sm-right\\">
                                        Have an account? 
                                        <Link to=\\"/login\\">
                                          <LinkAnchor href=\\"/login\\" navigate={[Function: navigate]}>
                                            <a href=\\"/login\\" onClick={[Function: onClick]}>
                                              Sign in here
                                            </a>
                                          </LinkAnchor>
                                        </Link>
                                      </small>
                                    </div>
                                  </Col>
                                </div>
                              </Row>
                              <Row className=\\"divider\\" tag=\\"div\\">
                                <div className=\\"divider row\\">
                                  <Col xs={12} sm={4} tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col-12 col-sm-4\\">
                                      <h5>
                                        Personal Details
                                      </h5>
                                    </div>
                                  </Col>
                                  <Col xs={12} sm={6} tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col-12 col-sm-6\\">
                                      <FormGroup tag=\\"div\\">
                                        <div className=\\"form-group\\">
                                          <Label tag=\\"label\\" widths={{...}}>
                                            <label htmlFor={[undefined]} className=\\"\\">
                                              Full Name
                                            </label>
                                          </Label>
                                          <input name=\\"name\\" />
                                        </div>
                                      </FormGroup>
                                    </div>
                                  </Col>
                                </div>
                              </Row>
                              <Row className=\\"divider\\" tag=\\"div\\">
                                <div className=\\"divider row\\">
                                  <Col xs={12} sm={4} tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col-12 col-sm-4\\">
                                      <h5>
                                        Login Details
                                      </h5>
                                    </div>
                                  </Col>
                                  <Col xs={12} sm={6} tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col-12 col-sm-6\\">
                                      <FormGroup tag=\\"div\\">
                                        <div className=\\"form-group\\">
                                          <Label tag=\\"label\\" widths={{...}}>
                                            <label htmlFor={[undefined]} className=\\"\\">
                                              Email
                                            </label>
                                          </Label>
                                          <input name=\\"email\\" />
                                        </div>
                                      </FormGroup>
                                      <FormGroup tag=\\"div\\">
                                        <div className=\\"form-group\\">
                                          <Label tag=\\"label\\" widths={{...}}>
                                            <label htmlFor={[undefined]} className=\\"\\">
                                              Password
                                            </label>
                                          </Label>
                                          <input name=\\"password\\" type=\\"password\\" />
                                        </div>
                                      </FormGroup>
                                    </div>
                                  </Col>
                                </div>
                              </Row>
                              <Row className=\\"divider footer\\" tag=\\"div\\">
                                <div className=\\"divider footer row\\">
                                  <Col tag=\\"div\\" widths={{...}}>
                                    <div className=\\"col\\">
                                      <div className=\\"btn-group float-right\\">
                                        <Button color=\\"success\\" outline={true} block={true} type=\\"submit\\" tag=\\"button\\">
                                          <button type=\\"submit\\" className=\\"btn btn-outline-success btn-block\\" onClick={[Function: bound onClick]} aria-label={{...}}>
                                            Register
                                          </button>
                                        </Button>
                                      </div>
                                    </div>
                                  </Col>
                                </div>
                              </Row>
                            </div>
                          </CardBody>
                        </div>
                      </Card>
                    </form>
                  </Form>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Register>
  </Router>
</MemoryRouter>"
`;
